### ISSUE TITLE
Optimize for completion with ADAM instead of gradient descent

### ISSUE THREAD

**Issue by bamos** [bamos/dcgan-completion.tensorflow#9]
+ [Algorithm 1 of this paper](https://arxiv.org/abs/1412.6980)
+ [Torch implementation for reference](https://github.com/torch/optim/blob/master/adam.lua)
